src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -31,17 +31,29 @@ public FindCommand parse(String args) throws ParseException {
 
         String[] nameKeywords = trimmedArgs.split("\\s+");
         List<String> fields = Arrays.asList(nameKeywords);
+        boolean isAllNumbers = true;
 
-        if (((int) (fields.get(0).charAt(0)) >= 65) & ((int) (fields.get(0).charAt(0)) <= 122)) {
+        for (int i = 0; i < fields.size(); i = i + 1) {
+            for (int j = 0; j < fields.get(i).length(); j = j + 1) {
+                if (fields.get(i).charAt(0) == 45) {
+                    throw new ParseException(String.format(MESSAGE_INVALID_ID_LENGTH_AND_SIGN,
+                            FindCommand.MESSAGE_USAGE));
+                }
```

Comment:
This check is only done once every i, so can put this outside the inner for loop i think. slightly better performance